verilog/tools/ls/verilog-language-server_test.cc

```diff
@@ -1442,6 +1442,138 @@ TEST_F(VerilogLanguageServerSymbolTableTest, CheckReferenceUnknownSymbol) {
   ASSERT_EQ(response_b["result"].size(), 0);
 }
 
+struct RenameRequestParams {
+  int id;
+  std::string file;
+  std::string newName;
+  int line;
+  int character;
+};
+
+std::string RenameRequest(RenameRequestParams params) {
+  json renamerq = {
+
+      {"jsonrpc", "2.0"},
+      {"id", params.id},
+      {"method", "textDocument/rename"},
+      {"params",
+       {
+           {"textDocument",
+            {
+                {"uri", params.file},
+            }},
+           {
+               "position",
+               {{"line", params.line}, {"character", params.character}},
+           },
+           {"newName", params.newName},
+       }}};
+  return renamerq.dump();
+}
+std::string PrepareRenameRequest(RenameRequestParams params) {
+  json renamerq = {
+
+      {"jsonrpc", "2.0"},
+      {"id", params.id},
+      {"method", "textDocument/prepareRename"},
+      {"params",
+       {
+           {"textDocument",
+            {
+                {"uri", params.file},
+            }},
+           {
+               "position",
+               {{"line", params.line}, {"character", params.character}},
+           },
+           {"newName", params.newName},
+       }}};
+  return renamerq.dump();
+}
+
+// Runs tests for textDocument/rangeFormatting requests
+TEST_F(VerilogLanguageServerSymbolTableTest, PrepareRenameTest) {
+  // Create sample file and make sure diagnostics do not have errors
+  RenameRequestParams params;
+  params.line = 2;
+  params.character = 1;
+  params.file = "file://" + root_dir + "/fmt.sv";
+  params.newName = "foo";
+  params.id = 1;
+  const std::string mini_module =
+      DidOpenRequest(params.file,
+                     "module fmt();\nfunction automatic "
+                     "bar();\nbar();\nbar();\nendfunction;\nendmodule\n");
+  ASSERT_OK(SendRequest(mini_module));
+
+  const json diagnostics = json::parse(GetResponse());
+  EXPECT_EQ(diagnostics["method"], "textDocument/publishDiagnostics")
+      << "textDocument/publishDiagnostics not received";
+  EXPECT_EQ(diagnostics["params"]["uri"], params.file)
+      << "Diagnostics for invalid file";
+
+  EXPECT_EQ(diagnostics["params"]["diagnostics"].size(), 0)
+      << "The test file has errors";
+  std::string request = PrepareRenameRequest(params);
+  ASSERT_OK(SendRequest(request));
+
+  const json response = json::parse(GetResponse());
+  EXPECT_EQ(response["result"]["start"]["line"], 2)
+      << "Invalid result for id:  ";
+  EXPECT_EQ(response["result"]["start"]["character"], 0)
+      << "Invalid result for id:  ";
+  EXPECT_EQ(response["result"]["end"]["line"], 2) << "Invalid result for id:  ";
+  EXPECT_EQ(response["result"]["end"]["character"], 3)
+      << "Invalid result for id:  ";
+}
+
+TEST_F(VerilogLanguageServerSymbolTableTest, RenameTest) {
```

Comment:
Multiple filenames will actually be interesting. So maybe have a test `RenameTestSymbolSingleFile` and `RenameTestSymbolMultipleFiles`, in which a symbol is defined in one file and used in another file - and the rename properly touches all files.

Reply:
Both are now renamed/added